### DISCUSSION TITLE
Unable to install provider package when latest crossplane version is installed.

### DISCUSSION THREAD

**Discussion by HRSanjay** [crossplane/crossplane#3680] [Show and tell]
We have installed crossplane by overriding the name as shown below. Crossplane installation was successful.

#values for crossplane
crossplane:
  nameOverride: ncom-crossplane-custom
  
Below are the serviceaccount's created as part of crossplane installation.
[cbnd@sanhr-dev-57 ~]$ kubectl get sa -n sanhr
NAME                                  SECRETS   AGE
default                               1         3d2h
rbac-manager                          1         3m15s
ncom-crossplane-custom   1         3m15s

When we tried to install aws-provider package, it failed with below error
Events:
  Type     Reason         Age                      From                                 Message
  ----     ------         ----                     ----                                 -------
  Warning  UnpackPackage  <invalid> (x7 over 58s)  packages/provider.pkg.crossplane.io  cannot unpack package: failed to fetch package digest from remote: serviceaccounts "crossplane" not found.

Why aws-provider package is expecting serviceAcccountName as "crossplane"? 

We have not faced any issues in older crossplane versions like 1.9.1 and 1.9.0.

This behavior was observed only on latest crossplane versions ( 1.10.0 and 1.10.1 ). This behavior is impacting aws provider package installation.

**Comment by hasheddan** [comment]
@HRSanjay if you override the `ServiceAccount` name, you can point Crossplane to it by specifying the name in [the `--service-account` flag](https://github.com/crossplane/crossplane/blob/ab78592350eed8e14c0d86f21800b8c53985b4e8/cmd/crossplane/core/core.go#L67). This can be accomplished using the [`args`](https://github.com/crossplane/crossplane/blob/ab78592350eed8e14c0d86f21800b8c53985b4e8/cluster/charts/crossplane/values.yaml.tmpl#L25) value in the Helm chart.

**Reply by HRSanjay** [reply]
Hi @hasheddan,

![image](https://user-images.githubusercontent.com/32261767/215500203-336050ca-5868-4557-86b7-2f65113368ac.png)

[cbnd@sanhr-dev-57 bcmt]$ kubectl get pods -n sanhr
NAME                                                              READY   STATUS    RESTARTS   AGE
sanhr-nc3102-ncom-crossplane-custom-845498977d-tbt4j              1/1     Running   0          7d
sanhr-nc3102-ncom-crossplane-custom-845498977d-vl7wd              1/1     Running   0          7d
sanhr-nc3102-ncom-crossplane-custom-rbac-manager-88949489b99d8d   1/1     Running   0          7d
sanhr-nc3102-ncom-crossplane-custom-rbac-manager-88949489bg9fhp   1/1     Running   0          7d
[cbnd@sanhr-dev-57 bcmt]$
[cbnd@sanhr-dev-57 bcmt]$
[cbnd@sanhr-dev-57 bcmt]$ kubectl get pkg -n sanhr
NAME                                                                                 INSTALLED   HEALTHY   PACKAGE                                                                               AGE
provider.pkg.crossplane.io/sanhr-nc3102-ncom-crossplane-custom-provider-aws                                registry1-docker-io.repo.cci.nokia.net/crossplane/provider-aws:v0.32.0                7d
provider.pkg.crossplane.io/sanhr-nc3102-ncom-crossplane-custom-provider-helm                               registry1-docker-io.repo.cci.nokia.net/crossplanecontrib/provider-helm:v0.12.0        7d
provider.pkg.crossplane.io/sanhr-nc3102-ncom-crossplane-custom-provider-kubernetes                         registry1-docker-io.repo.cci.nokia.net/crossplanecontrib/provider-kubernetes:v0.5.0   7d
provider.pkg.crossplane.io/sanhr-nc3102-ncom-crossplane-custom-provider-terraform                          registry1-docker-io.repo.cci.nokia.net/crossplanecontrib/provider-terraform:v0.6.0    7d
[cbnd@sanhr-dev-57 bcmt]$
[cbnd@sanhr-dev-57 bcmt]$
[cbnd@sanhr-dev-57 bcmt]$
[cbnd@sanhr-dev-57 bcmt]$ helm list -n sanhr
NAME                                    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                           APP VERSION
sanhr-nc3102-ncom-crossplane-custom     sanhr           1               2023-01-23 19:29:14.883825786 +0530 IST deployed        crossplane-custom-0.0.0-301     1.0

![image](https://user-images.githubusercontent.com/32261767/215501889-09649fbf-2df9-49a6-913f-ee5e1c41fedf.png)

No issues w.r.t crossplane installation.  provider packages like aws, helm,etc - we couldn't able to install it.

How can be fix provider package installation?


if you override the ServiceAccount name, you can point Crossplane to it by specifying the name in [the --service-account flag]   -->  this is for crossplane microservice.   How can we point it to provider packages.

**Reply by hasheddan** [reply]
> if you override the ServiceAccount name, you can point Crossplane to it by specifying the name in [the --service-account flag] --> this is for crossplane microservice. How can we point it to provider packages.

@HRSanjay the flag is to inform Crossplane of which `ServiceAccount` to use when pulling packages.

**Reply by HRSanjay** [reply]
Hi @hasheddan

![image](https://user-images.githubusercontent.com/32261767/215516047-a92a133d-bc20-4c40-a76d-4ed5597daf6b.png)

Only these are the supported flags, I don't find service account flag in the above list.   Can you please suggest the syntax for service account usage in args.

#values for crossplane
crossplane:
  nameOverride: sanhr-nc3102-ncom-crossplane-custom
  replicas: 2
  args: [ "--service-account=sanhr-nc3102-ncom-crossplane-custom" ]
  nodeSelector: {}
  tolerations: {}
  affinity: {}

**Reply by hasheddan** [reply]
@HRSanjay this flag was added in `v1.10.0` -- do you happen to be on an older version? https://github.com/crossplane/crossplane/commit/c2a6242bba1aa72813a2fc1b8c10455ffb2dd9cc

**Reply by HRSanjay** [reply]
It worked. Thanks @hasheddan for the support.
